File: programmers/js/new/level2/숫자의_표현.js
function solution(n) {
    var answer = 0;
    let left = 1;
    let right = 2;
    let sum = 1;
    while (left < right) {
        if (sum === n) {
            answer++;
            sum -= left;
            left++;
        } else if (sum < n) {
            sum += right;
            right++;
        } else if (sum > n) {
            sum -= left;
            left++;
        }
    }
    return answer;
}

console.log(solution(15));

File: programmers/python/level2/숫자의_표현.py
def solution(n):
    answer = 0
    left = 1
    right = 1
    my_sum = 1
    while left <= n:
        if my_sum == n:
            answer += 1
            right += 1
            my_sum += right
        elif my_sum > n:
            my_sum -= left
            left += 1
        elif my_sum < n:
            right += 1
            my_sum += right
    return answer

print(solution(15))